nautobot/extras/jobs.py

```diff
@@ -979,21 +990,23 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
         job_result.completed = timezone.now()
         job_result.save()
         return False
-    job = job_class()
-    job.job_result = job_result
 
-    # Capture the file IDs for any FileProxy objects created so we can cleanup later.
-    file_fields = list(job._get_file_vars())
-    file_ids = [data[f] for f in file_fields]
+    try:
+        job = job_class()
```

Comment:
Consider the possibility that `job_class()` could itself raise an exception, in which case `job` would be undefined in the `except` block.